src/modules/payload_deliverer/payload_deliverer.cpp

```diff
@@ -151,22 +150,86 @@ void PayloadDeliverer::handle_vehicle_command(const hrt_abstime &now,  const veh
+
+		// Flag to indicate if we can process the new gripper command
+		bool process_current_gripper_cmd{false};
+
+		// We are currently in the middle of executing a previous vehicle command. Handle the conflicts
+		if (_cur_vcmd_gripper_action != GRIPPER_ACTION_NONE) {
 
-		switch (gripper_action) {
-		case vehicle_command_s::GRIPPER_ACTION_GRAB:
-			_gripper.grab();
-			break;
+			if (gripper_action != _cur_vcmd_gripper_action) {
+				// If different action is commanded, cancel the previous vehicle command
+				send_gripper_vehicle_command_ack(now, vehicle_command_ack_s::VEHICLE_CMD_RESULT_CANCELLED, _cur_vcmd_target_system,
+								 _cur_vcmd_target_component);
 
-		case vehicle_command_s::GRIPPER_ACTION_RELEASE:
-			_gripper.release();
-			break;
+				// NOTE: Does this make sense when the same entity sends two conflicting commands?
+				// For the previous command, the external entity will received CANCELLED result, but then
+				// would receive IN_PROGRESS for the new command. As the COMMAND_CANCEL isn't defined properly
+				// in MAVLink yet(https://mavlink.io/en/messages/common.html#COMMAND_CANCEL), this can lead to
+				// Ground stations to thinking it's new command has been canceled instead of the previous one!
+
+				process_current_gripper_cmd = true;
+
+			} else {
+				// If same action is commanded, temporarily reject current command (since it's already in progress)
+				send_gripper_vehicle_command_ack(now, vehicle_command_ack_s::VEHICLE_CMD_RESULT_TEMPORARILY_REJECTED,
+								 vehicle_command->source_system, vehicle_command->source_component);
+			}
+
+		} else {
+			// If we aren't processing any gripper commands, always process the new command
+			process_current_gripper_cmd = true;
+
+		}
+
+		// Flag to indicate if we executed the current gripper command
+		bool current_gripper_command_executed{false};
+
+		if (process_current_gripper_cmd) {
+			if ((_gripper.grabbed() && _cur_vcmd_gripper_action == vehicle_command_s::GRIPPER_ACTION_GRAB) ||
+			    (_gripper.released() && _cur_vcmd_gripper_action == vehicle_command_s::GRIPPER_ACTION_RELEASE)) {
+				// First check if we already satisfied the command. If so, acknowledge as accepted and don't execute the command
+				send_gripper_vehicle_command_ack(now, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED,
+								 vehicle_command->source_system, vehicle_command->source_component);
+
+			} else {
+				// Check if the gripper action is valid
+				switch (gripper_action) {
+				case vehicle_command_s::GRIPPER_ACTION_GRAB:
+					_gripper.grab();
+					current_gripper_command_executed = true;
+					break;
+
+				case vehicle_command_s::GRIPPER_ACTION_RELEASE:
+					_gripper.release();
+					current_gripper_command_executed = true;
+					break;
+
+				default:
+					// INVALID GRIPPER ACTION. Deny the command.
+					send_gripper_vehicle_command_ack(now, vehicle_command_ack_s::VEHICLE_CMD_RESULT_DENIED, vehicle_command->source_system,
```

Comment:
```suggestion
					send_gripper_vehicle_command_ack(now, vehicle_command_ack_s::VEHICLE_CMD_RESULT_UNSUPPORTED, vehicle_command->source_system,
```

Reply:
Doesn't it fit DENIED description better? That is:
`Command is invalid (is supported but has invalid parameters). Retrying same command and parameters will not work.`

rather than UNSUPPORTED:
`Command is not supported (unknown).`

Since the command (DO_GRIPPER) it supported, but the parameters are invalid?

https://mavlink.io/en/messages/common.html#MAV_RESULT

Reply:
The enum might get expanded in the future with new commands.

Reply:
Please address this

Reply:
Alright DENIED is correct according to the mavlink spec. I certainly wouldn't have expected that.